### ISSUE TITLE
Add accepted() and current() to MCMCSampler or update return type of tell()

### ISSUE THREAD

**Issue by MichaelClerx** [pints-team/pints#660]
At various places in the code we need to know more from a sampler than just the current position it is at.
I.E.

- when storing evaluations inside the controller, we need to know the evaluation for the accepted sample (which is not necessarily the last sample that was evaluated)
- inside population MCMC we need to know whether or not a tell() accepted the sample, to update our internal mirror of its state
- for some model selection methods, I've been told

My initial proposal was to add methods `accepted() -> bool` that tell you what happened on the last step, and `current()` that would give the current position + evalution possibly.

Now thinking it might be better to have this as part of tell. E.g.
```
class MCMCSampler
  def tell(fx):
    ...
    return current, current_log_pdf, accepted_yes_or_no
```
or
```
class MCMCSampler
  def tell_verbose(fx)
    ...
    return current, current_log_pdf, accepted_yes_or_no
  def tell(fx)
    return self.tell_verbose(fx)[0]
```
or
```
class MCMCSampler
  def tell(fx, verbose=False)
    if verbose:
      return current, current_log_pdf, accepted_yes_or_no
    return current
```

My preference would be for 1 or 2, I think.

Thoughts on this much appreciated @ben18785 @martinjrobins @chonlei @fcooper8472 

---------------------------------------------------------------------------------------------------------------------------------------------------------

Original message:

*Add accepted() and current() to MCMCSampler*

To get more info about what happened on the last step. This makes it easier to use MCMC samplers inside another method

**Comment by MichaelClerx** [comment]
@martinjrobins ?

We need these e.g. in https://github.com/pints-team/pints/pull/664/files which currently uses hacky methods not guaranteed to work

**Comment by MichaelClerx** [comment]
Might allow `current_log_pdf` to be dropped

**Comment by MichaelClerx** [comment]
Closed in #1250
